shell/platform/linux/fl_renderer_headless.cc

```diff
@@ -0,0 +1,29 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "fl_renderer_headless.h"
+
+struct _FlRendererHeadless {
+  FlRenderer parent_instance;
+};
+
+G_DEFINE_TYPE(FlRendererHeadless, fl_renderer_headless, fl_renderer_get_type())
+
+static EGLSurface fl_renderer_headless_create_surface(FlRenderer* renderer,
+                                                      EGLDisplay display,
+                                                      EGLConfig config) {
+  return EGL_NO_SURFACE;
+}
+
+static void fl_renderer_headless_class_init(FlRendererHeadlessClass* klass) {
+  FL_RENDERER_CLASS(klass)->create_surface =
+      fl_renderer_headless_create_surface;
+}
+
+static void fl_renderer_headless_init(FlRendererHeadless* self) {}
+
```

Comment:
Why doesn't this class need to override all the callbacks to do nothing and return false? Isn't, e.g., `fl_renderer_make_current` going to have issues when there's no surface?

Reply:
`FlRenderer` only has the one pure virtual method; `fl_renderer_make_current` is implemented inside the `FlRenderer` class. We decided on review that there was no need for renderer subclasses to override this.

The current implementation of `fl_renderer_make_current` and `fl_renderer_present` don't explicitly return false when the surface is null, but the associated EGL call will return an error with the same effect. I figure that Flutter should know there is no surface and not call these (it doesn't appear to). If it did it would be useful to get the error message.